lib/src/notification.dart

```diff
@@ -349,6 +391,10 @@ Future<void> registerAndroidPushNotificationHandlers(
 
   _androidSettings = androidSettings;
 
+  // Save push notification settings to shared preferences.
+  final preferences = await SharedPreferences.getInstance();
+  preferences.setString(ANDROID_SETTINGS, jsonEncode(_androidSettings));
+
```

Comment:
Is there a possibility that 2 notifications arrive before they are handled, so that the `ANDROID_SETTINGS` string is overwritten, thus handling only the second notification?
If so, it would be better to have a list, and append to that, right?

Reply:
No, it's not possible. `registerAndroidPushNotificationHandlers` is only called on startup inside the `main` function. When the app has been terminated and a notification arrives, Dart just starts a service called `FlutterFirebaseMessagingBackgroundService` if it is not already running. This service then only calls the handler we passed to `FirebaseMessaging.onBackgroundMessage` below.